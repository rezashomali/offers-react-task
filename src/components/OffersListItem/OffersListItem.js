import { makeStyles } from "@material-ui/core/styles";
import {
  Grid,
  Card,
  CardActions,
  CardContent,
  CardMedia,
  Typography,
} from "@material-ui/core";

const useStyles = makeStyles({
  carName: {
    height: "40px",
  },
  carImage: {
    width: "80%",
    margin: "auto",
    height: "auto",
  },
});

const OffersListItem = ({ offer }) => {
  const classes = useStyles();
  return (
    <Grid item xs={12} sm={6} md={4}>
      <Card>
        <CardMedia
          component="img"
          className={classes.carImage}
          alt={offer.name}
          height="120"
          image={
            offer.image
              ? `https://www.sixt.de${offer.image}`
              : "https://www.sixt.com/fileadmin/files/global/user_upload/fleet/png/350x200/iveco-daily-kasten-weiss-sx-2015.png"
          }
          title={offer.name}
        />
        <CardContent>
          <Typography
            gutterBottom
            variant="h6"
            component="h2"
            className={classes.carName}
          >
            {offer.name}
          </Typography>
        </CardContent>

        <CardActions>
          <Typography variant="subtitle2" color="textSecondary" component="h2">
            {`â‚¬ ${offer.price} day`}
          </Typography>
        </CardActions>
      </Card>
    </Grid>
  );
};

export default OffersListItem;
