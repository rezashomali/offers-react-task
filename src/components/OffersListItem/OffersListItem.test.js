import { render } from "@testing-library/react";
import OffersListItem from "./OffersListItem";

test("offersListItem snapshot", () => {
  const data = {
    id: "2436432a-e1d5-4649-bece-f98834f6ee1d-XJAR",
    image:
      "//fileadmin/files/global/user_upload/fleet/png/630x360/bmw-x7-m50d-5d-white-2019.png",
    name: "BMW X7 M50",
    price: 797.97,
  };
  const container = render(<OffersListItem offer={data} />);
  expect(container.firstChild).toMatchSnapshot();
});
