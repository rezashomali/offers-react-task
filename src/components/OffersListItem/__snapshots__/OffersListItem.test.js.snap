// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`offersListItem snapshot 1`] = `undefined`;
